c/secp256k1_blake160_multisig_all.c

```diff
@@ -0,0 +1,220 @@
+#include "blake2b.h"
+#include "ckb_syscalls.h"
+#include "protocol_reader.h"
+#include "secp256k1_helper.h"
+
+#define ERROR_ARGUMENTS_LEN -1
+#define ERROR_MULTSIG_SCRIPT_HASH -2
+#define ERROR_SYSCALL -3
+#define ERROR_VERIFICATION -4
+#define ERROR_SECP_RECOVER_PUBKEY -5
+#define ERROR_SECP_PARSE_SIGNATURE -6
+#define ERROR_SECP_SERIALIZE_PUBKEY -7
+#define ERROR_ENCODING -8
+#define ERROR_WITNESS_TOO_LONG -9
+#define ERROR_WITNESS_TOO_SHORT -10
+#define ERROR_INVALID_PUBKEYS_CNT -11
+#define ERROR_INVALID_THRESHOLD -12
+#define ERROR_INVALID_REQUIRE_FIRST_N -13
+
+#define BLAKE2B_BLOCK_SIZE 32
+#define BLAKE160_SIZE 20
+#define PUBKEY_SIZE 33
+#define TEMP_SIZE 1024
+#define RECID_INDEX 64
+/* 32 KB */
+#define WITNESS_SIZE 32768
+#define SCRIPT_SIZE 32768
+#define SIGNATURE_SIZE 65
+#define FLAGS_SIZE 4
+
+/*
+ * Arguments:
+ * multisig script blake160 hash, 20 bytes.
+ *
+ * Witness:
+ * multisig_script | Signature1 | signature2 | ...
+ * multisig_script: S | R | M | N | Pubkey1 | Pubkey2 | ...
```

Comment:
The script doesn't need to store the pubkey. Pubkey hash is enough since you are using recoverable signature.

Reply:
This script stores pubkey in multisig script to reduce cycle.

Reply:
On the other hand, use pubkey hash allows shorter witness size, hence shorter transaction size. We don't know yet in the longer run, which will be more important. Another reason for using a real serialization solution as mentioned below.